desk.py

```diff
@@ -1,88 +1,100 @@
+    """
+
+    def __init__(self, desk_name, desk_id=-1, occupied=False):
+        """
+        Initialize a new desk object
+        :param desk_name: Name of desk
+        :param desk_id: ID of desk (associated with server)
+        :param occupied: Boolean of occupancy status
+        """
+
         self.__desk_name = desk_name
         self.__occupied = occupied
-        self.__id = id
+        self.__id = desk_id
         self.__desk_observers = set()
 
-    '''
-    Print Desk Info  *such wow*
-    '''
-    def print_info(self):
-        print("Desk Name:", self.__desk_name, ", Currently Occupied:", self.__occupied)
-
-    '''
-    Gets whether or not this desk object is currently occupied
-    '''
-    def get_occupied(self):
-        with self.__lock:
-            return self.__occupied
-
-    '''
-    Get Desk ID
-    '''
-    def get_id(self):
+    def __str__(self):
+        """
+        Return Desk Info  *such wow*
+        :return: String
+        """
+
+        return "Desk Name: '" + self.__desk_name + "', Currently Occupied: " + self.__occupied
+
+    @property
+    def occupied(self):
+        """
+        :return: Whether or not this desk object is currently occupied
+        """
+
+        return self.__occupied
+
+    @property
+    def desk_id(self):
+        """
+        :return: Desk ID
+        """
+
         return self.__id
 
-    '''
-    Get Desk Name
-    '''
-    def get_name(self):
+    @property
+    def name(self):
+        """
+        :return: Desk Name
+        """
+
         return self.__desk_name
 
-    '''
-    Register an observer for occupancy events
-    '''
-    def register_occupied_change_event(self, obj):
-        if obj not in self.__desk_observers:
-            self.__desk_observers.add(obj)
+    def add_observer(self, obj):
+        """
+        Register an observer for occupancy events. Must implement the
+        DeskObserver protocol
+        :param obj: The object that will subscribe to desk events
+        :return: None
+        """
 
-    '''
-    Register an observer for occupancy events
-    '''
-    def unregister_occupied_change_event(self, obj):
-        self.__desk_observers.remove(obj)
+        if isinstance(obj, DeskObserver):
+            self.__desk_observers.add(obj)
 
-    '''
-    Input Received from GPIO pins
-    '''
-    def input_received(self, pin, value):
+    def remove_observer(self, obj):
+        """
+        Stop an observer from receiving desk occupancy events
+        :param obj: The object that should be removed
+        :return: None
+        """
 
-        # Use a thread lock for synchronizing this method so that no weird behaviour occurs
-        # This also ensures that all subscribers get notified correctly and in order
-        with self.__lock:
-            self.__occupied = value == 1
-            print("Desk Occupied: ", value)
+        if obj in self.__desk_observers:
+            self.__desk_observers.remove(obj)
```

Comment:
You can use `set#discard` to perform this operation in one line and with only 1 traversal of `__desk_observers`